common/src/main/java/com/microsoft/identity/common/internal/platform/IDevicePopManager.java

```diff
@@ -146,6 +146,35 @@ public String toString() {
         }
     }
 
+    /**
+     * Information about the backing of the underlying keystore.
+     */
+    enum SecureHardwareState {
+        /**
+         * Return if the underlying private key resides inside secure hardware (e.g., Trusted
+         * Execution Environment (TEE) or Secure Element (SE)). No mechanism of attestation is
+         * provided or specified.
+         */
+        TRUE_UNATTESTED,
+
+        /**
+         * The the underlying private key is not inside secure hardware.
+         */
+        FALSE,
+
+        /**
+         * It is unknown where the underlying key resides, due to lack of API support for
+         * determination.
+         */
+        UNKNOWN_DOWNLEVEL,
+
+        /**
+         * It is unknown where the underlying key resides, due to an error during keystore
+         * interrogation.
+         */
+        UNKNOWN_QRY_ERR
```

Comment:
What is `QRY`?

Reply:
It's query -- @AdamBJohnsonx had same question -- will rename for clarity

https://github.com/AzureAD/microsoft-authentication-library-common-for-android/pull/1069#discussion_r500459410

Reply:
910b23617bc37c764902aade03e990844ea7ef06